examples/kubernetes/deployment.yaml

```diff
@@ -21,7 +21,7 @@ spec:
     spec:
       containers:
         - args:
-            - /usr/local/bin/synthetic-monitoring-agent --api-server-address=${API_SERVER} --api-token=${API_TOKEN} --verbose=true
+            - /usr/local/bin/synthetic-monitoring-agent --api-server-address=${API_SERVER} --api-token=${API_TOKEN} --verbose=true --listen-address=0.0.0.0:4050
```

Comment:
```suggestion
            # Please note that by listening on all interfaces (passing 0.0.0.0 to --listen-address) anyone with access
            # to the cluster might be able to access information about the agent that might not be available otherwise.
            # The HTTP server is used to present metrics as well as some control and debugging endpoints.
            - /usr/local/bin/synthetic-monitoring-agent --api-server-address=${API_SERVER} --api-token=${API_TOKEN} --verbose=true --listen-address=0.0.0.0:4050
```

Reply:
I understand the reason behind adding a warning, but I'm not 100% sure about its shape. I'm afraid users might see it and read it as a "this default is dangerous for some reason and you should change it!". Then, if they do, the deployment breaks.

Do we think this information leak is dangerous enough to warn about it? In my experience as a k8s admin, I mostly expect pods to listen in `0.0.0.0` by default. If I'm worried about that, I should deploy some network policies to harden it.